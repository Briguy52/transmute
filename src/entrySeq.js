import { Iterable, Seq } from "immutable";

/**
 * Get a Seq of the entries (i.e. [key, value] tuples) in `subject`.
 *
 * @param  {Array|Iterable|Object} subject
 * @return {Seq}
 */
export default function entrySeq(subject) {
  if (!Iterable.isIterable(subject)) {
    subject = Seq(subject);
  }
  return subject.entrySeq();
}
