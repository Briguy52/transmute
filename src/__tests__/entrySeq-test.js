import { Map } from "immutable";
import entrySeq from "../entrySeq";

describe("transmute/entrySeq", () => {
  it("gets the values from a Map", () => {
    const entries = entrySeq(Map({ one: 1, two: 2, three: 3 }));
    expect(entries).toMatchSnapshot();
  });

  it("gets the entries from an Object", () => {
    const res = entrySeq({ one: 1, two: 2, three: 3 });
    expect(res).toMatchSnapshot();
  });
});
