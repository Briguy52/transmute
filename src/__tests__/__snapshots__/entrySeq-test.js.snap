// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/entrySeq gets the entries from an Object 1`] = `
Array [
  Array [
    "one",
    1,
  ],
  Array [
    "two",
    2,
  ],
  Array [
    "three",
    3,
  ],
]
`;

exports[`transmute/entrySeq gets the values from a Map 1`] = `
Array [
  Array [
    "one",
    1,
  ],
  Array [
    "two",
    2,
  ],
  Array [
    "three",
    3,
  ],
]
`;
